test/app/model/ChessEngine.spec.ts

```diff
@@ -108,4 +108,56 @@ describe('Chess Engine', () => {
 
     expect(() => engine.move({ column: 'A', row: 2 }, { column: 'A', row: 6 })).toThrowError('There is no piece on this square.');
   });
+
+  it('After white players move a black king is in white piece range, then black king is checked', () => {
+    const whitePiece = new Bishop(Side.WHITE);
+    const blackPiece = new King(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { D8: blackPiece, F4: whitePiece };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const bishopSquareFrom: Square = { column: 'F', row: 4 };
+    const bishopSquareTo: Square = { column: 'G', row: 5 };
+
+    expect(engine.move(bishopSquareFrom, bishopSquareTo)).toIncludeSameMembers([
+      {
+        eventType: 'PieceWasMoved',
+        piece: whitePiece,
+        from: bishopSquareFrom,
+        to: bishopSquareTo,
+      },
+      {
+        eventType: 'KingWasChecked',
+        king: blackPiece,
+        onSquare: { column: 'D', row: 8 },
+      },
+    ]);
+  });
+
+  it('After black players move, black king is no longer checked', () => {
+    const whitePiece = new Bishop(Side.WHITE);
+    const blackPiece = new King(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { D8: blackPiece, F4: whitePiece };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const kingSquareFrom: Square = { column: 'D', row: 8 };
+    const kingSquareTo: Square = { column: 'D', row: 7 };
+    const bishopSquareFrom: Square = { column: 'F', row: 4 };
+    const bishopSquareTo: Square = { column: 'G', row: 5 };
+
+    engine.move(bishopSquareFrom, bishopSquareTo);
+
+    expect(engine.move(kingSquareFrom, kingSquareTo)).toIncludeSameMembers([
```

Comment:
To będzie troszkę trudniejsze do przetestowania, ale przydałby się taki test case:
- WHEN robie ruch - THEN król jest szachowany
- WHEN robie ruch, który ochrania króla - THEN król nie jest szachowany

Bo teraz event KingWasUnchecked będzie za każdym ruchem z tego co widzę, co nie szachuje króla. A by przydało się tylko, jeśli król wcześniej był szachowany.
Tutaj napotkałeś problem z setupem testu. Ale może move zrobić jako setup :) 

EDIT: Aaa, widze, ze to zrobiles :) Bardzo ładnie, że to wykminiles!